hs_core/views/__init__.py

```diff
@@ -689,6 +689,7 @@ def set_resource_flag(request, shortkey, *args, **kwargs):
     elif flag == 'make_private' or flag == 'make_not_discoverable':
         _set_resource_sharing_status(request, user, res, flag_to_set='discoverable', flag_value=False)
     elif flag == 'make_discoverable':
+        _set_resource_sharing_status(request, user, res, flag_to_set='public', flag_value=False)
         _set_resource_sharing_status(request, user, res, flag_to_set='discoverable', flag_value=True)
```

Comment:
@aphelionz Instead of adding ```_set_resource_sharing_status(request, user, res, flag_to_set='public', flag_value=False)```, I think it is better to handle the logic inside ```set_discoverable()``` in ```hs_core\models.py``` which already insure if discoverable is set to False, public flag is set to False as well. I guess a logic can be added there to make sure if discoverable is set to True, public flag is set to False as well. See https://github.com/hydroshare/hydroshare/blob/develop/hs_core/models.py#L1693-L1694 for reference.